vendor/k8s.io/kubernetes/test/e2e/apps/cronjob.go

```diff
@@ -179,8 +179,8 @@ var _ = SIGDescribe("CronJob", func() {
 		Expect(err).NotTo(HaveOccurred())
 
 		By("Ensuring no unexpected event has happened")
-		err = checkNoEventWithReason(f.ClientSet, f.Namespace.Name, cronJob.Name, []string{"MissingJob", "UnexpectedJob"})
-		Expect(err).NotTo(HaveOccurred())
+		err = waitForEventWithReason(f.ClientSet, f.Namespace.Name, cronJob.Name, []string{"MissingJob", "UnexpectedJob"})
+		Expect(err).To(HaveOccurred())
```

Comment:
I am assuming that the change here is because we'll get an error after polling and not finding the event? Also, won't this end up adding ~30 extra seconds to the test? Does that matter?

Reply:
Yup, for both and that's ok-ish. Other option was to have two bits of code, which is hard to justify atm.